docs/cli-usage.md

```diff
@@ -0,0 +1,141 @@
+'instances' flag. A hanging-poll strategy is used, meaning that changes to the
+metadata value will be reflected in the `-dir` even while the proxy is running.
+When an instance is removed from the list the corresponding socket will be
+removed from `-dir` as well (unless it was also specified in `-instances`), but
+any existing connections to this instance will NOT be terminated.
+
+Note: `-instances` and `-instances_metadata` may be used at the same time but
+are not compatible with the `-fuse` flag.
+
+* `-ip_address_types=PUBLIC,PRIVATE`
+
+A comma-delimited list of preferred IP types for connecting to an instance. For
+example, setting this to PRIVATE will force the proxy to connect to instances
+using an instance's associated private IP. Defaults to `PUBLIC,PRIVATE`
+
+* `-term_timeout=30s`
+
+How long to wait for connections to close before shutting down the proxy.
+Defaults to 0.
+
+* `-skip_failed_instance_config`
+
+Setting this flag will allow you to prevent the proxy from terminating when some
+instance configurations could not be parsed and/or are unavailable.
+
+* `-log_debug_stdout=true`
+
+This is to log non-error output to stdOut instead of stdErr. For example, if you
+don't want connection related messages to log as errors, set this flag to true.
+Defaults to false.
+
+* `-credential_file`
+
+Allows use of the proxy outside of Google's cloud. Simply [create a new service
+account](https://cloud.google.com/sql/docs/mysql/sql-proxy#create-service-account),
+download the associated JSON file, and set `-credential_file` to the path of the
+JSON file. You can also set the GOOGLE_APPLICATION_CREDENTIALS environment
+variable instead of passing this flag.
+
+## Authentication
+
+Authentication can be configured in a few different ways. Those listed higher on
+the list will override options lower on the list:
+
+1. `credential_file` flag
+2. `token` flag
+3. Service account key at path stored in `GOOGLE_APPLICATION_CREDENTIALS`
+4. gcloud _user_ credentials (set from `gcloud auth login`)
+5. Default Application Credentials via goauth:
+
+   1. `GOOGLE_APPLICATION_CREDENTIALS` (again)
+   2. gcloud _application default_ credentials (set from ` gcloud auth application-default login`)
+   3. appengine.AccessToken (for App Engine Go < =1.9)
+   4. GCE/GAE metadata credentials
+
+Note: any account connecting to a Cloud SQL database will need the
+`roles/cloudsql.client` role. See [Roles and Permissions in Cloud
+SQL](https://cloud.google.com/sql/docs/mysql/roles-and-permissions) for details.
+
+When the proxy authenticates under the default service account of the Compute
+Engine VM it is running on the VM must have at least the sqlservice.admin API
+scope ("https://www.googleapis.com/auth/sqlservice.admin") and the associated
+project must have the SQL Admin API enabled.  The default service account must
+also have at least WRITER/EDITOR priviledges to any projects of target SQL
+instances.
+
+## Example Invocations
```

Comment:
At a more meta level, I prefer when the examples are next to the documentation for the flag itself. If someone just wants a list of flags they can use `-help`.

Reply:
I'll move all examples below their flags. I agree it's more useful.